core/components/minishop2/model/minishop2/mscarthandler.class.php

```diff
@@ -207,13 +209,19 @@ public function add($id, $count = 1, $options = array())
             if (array_key_exists($key, $this->cart)) {
                 return $this->change($key, $this->cart[$key]['count'] + $count);
             } else {
+                if ($this->modx->getOption('ms2_cart_context', null, '', true) == 1){
+                    $ctx_key = 'web';
+                }
+                else {
+                    $ctx_key = $this->modx->context->get('key');
+                }
```

Comment:
$ctx_key = 'web';
               if (!$this->modx->getOption('ms2_cart_context', null, '', true) ){
                    $ctx_key = $this->modx->context->get('key');
                }